domains/Discussions/Tests/Feature/QuestionsStoreTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Domains\Discussions\Tests\Feature;
+
+use Domains\Accounts\Database\Factories\UserFactory;
+use Domains\Accounts\Models\User;
+use Domains\Discussions\Database\Factories\QuestionFactory;
+use Domains\Discussions\Models\Question;
+use Faker\Factory;
+use Faker\Generator;
+use Illuminate\Http\Response;
+use Illuminate\Support\Str;
+use Laravel\Lumen\Testing\DatabaseMigrations;
+use Tests\TestCase;
+
+class QuestionsStoreTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    private Generator $faker;
+    private User $user;
+    private Question $question;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->faker = Factory::create();
+        $this->user = UserFactory::new()->create();
+        $this->question = QuestionFactory::new(['author_id' => $this->user->id])->create();
+    }
+
+    /** @test */
+    public function it_stores_questions(): void
+    {
+        $payload = [
+            'title' => $this->faker->title,
+            'description' => $this->faker->paragraph,
+        ];
+
+        $this->actingAs($this->user);
+        $response = $this->call('POST', '/questions', $payload);
+        self::assertTrue($response->isEmpty());
+
+        $this->seeInDatabase('questions', [
+            'author_id' => $this->user->id,
+            'title' => $payload['title'],
+            'description' => $payload['description'],
+        ]);
+    }
+
+    /** @test */
+    public function it_calculates_a_slug(): void
+    {
+        self::assertEquals(Str::slug($this->question->title), $this->question->slug);
+    }
```

Comment:
We could add this assertion in `it_stores_questions()` test case. It'll also remove the need to pre-create a Question resource for this, which I find it weird to happen.

Reply:
Ok